src/main/java/seedu/moneygowhere/model/budget/Budget.java

```diff
@@ -48,30 +74,125 @@ public static boolean isValidBudget(double test) {
+            this.month = temp;
+            this.sum = 0;
+            logger.info("BudgetMonth has been updated");
+        } else {
+            logger.info("BudgetMonth is not updated");
+        }
+    }
+
     public double getValue() {
         return value;
     }
 
+    public double getSum() {
+        return sum;
+    }
+
+    public String getValueString() {
+        return String.format("%.2f", value);
+    }
+
+    public String getSumString() {
+        return String.format("%.2f", sum);
+    }
+
+    public String getMonthString() {
+        return month.toString();
+    }
+
     public void setValue(double value) {
         this.value = value;
     }
 
+    public void clearBudgetSum() {
+        sum = 0;
+    }
+
+    /**
+     * Checks whether a spending is in the same month of the Budget.
+     * @param s the spending to check.
+     * @return whether or not the spending is in the same month as the budget
+     */
+    public boolean inSameMonth(Spending s) {
+        BudgetMonth temp = new BudgetMonth(s.getDateValue());
+        return temp.equals(this.month);
+    }
+
+    /**
+     * Adds the spending cost if it is in the same month.
+     * @param s The spending to add.
+     */
+    public void addSpending(Spending s) {
+        if (inSameMonth(s)) {
+            sum += Double.parseDouble(s.getCost().toString());
+        }
+    }
+
+    /**
+     * deletes the spending cost if it is in the same month.
+     * @param s The spending to delete.
+     */
+    public void deleteSpending(Spending s) {
+        if (inSameMonth(s)) {
+            sum -= Double.parseDouble(s.getCost().toString());
+        }
+    }
+
+    public double getRemainingBudget() {
+        return value - sum;
+    }
+
+    public void setBudget(Budget budget) {
+        this.value = budget.value;
+        this.sum = budget.sum;
+        this.month = budget.month;
+    }
+
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Budget // instanceof handles nulls
-                && value == ((Budget) other).value); // state check
+                && value == ((Budget) other).value
+                && sum == ((Budget) other).sum // state check
+                && month.equals(((Budget) other).month)); // state check
     }
 
     @Override
     public int hashCode() {
-        return Double.hashCode(value);
+        return Objects.hash(value, sum);
     }
 
     /**
      * Format state as text for viewing.
      */
     public String toString() {
-        return String.format("$%.2f", value);
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("$%.2f", value));
+        return sb.toString();
+    }
+
+    /**
+     * Return a complete description of the budget.
+     * @return A string representing the set budget only.
+     */
+    public String toStringFull() {
```

Comment:
This could be refactored into two methods to avoid duplication, so `toString` calls one with an argument to get the complete description and this one does not. `toStringFull` could be renamed to `getBudgetMessage`, something clearer